src/commands/command_parser.h

```diff
@@ -56,12 +60,39 @@ struct CommandParser {
 
   decltype(auto) RawPeek() const { return *begin_; }
 
+  decltype(auto) operator[](size_t index) const { return begin_[index]; }
```

Comment:
I think you can use something like std::advance here to avoid the requirement of random access iterator.